code/game/objects/effects/prop/snake.dm

```diff
@@ -0,0 +1,125 @@
+
+/obj/effect/temporary_effect/pulse/snake
+	name = "snake head"
+	pulses_remaining = 20
+	pulse_delay = 0.5 SECONDS
+
+	icon_state = "arrow_omni"
+
+	invisibility = 100
+
+// The atom which created this.
+	var/atom/movable/creator
+// Will the snake ever intentionally move onto its creator's turf?
+	var/safe = FALSE
+
+	var/ignore_density = FALSE
+
+// The turfs this snake has already crossed.
+	var/list/iterated_turfs = list()
+// How many turfs this snake should remember.
+	var/total_turf_memory = 5
+// Is the snake hunting a specific atom? (Will always try to meander toward this target.)
+	var/atom/hunting
+
+/obj/effect/temporary_effect/pulse/snake/New(var/turf/T, var/atom/hunt_target, var/atom/Creator)
+	if(hunt_target)
+		hunting = hunt_target
+
+	if(Creator)
+		creator = Creator
+
+	..()
+
+/obj/effect/temporary_effect/pulse/snake/pulse_loop()	// Override needed unfortunately to handle the possibility of not finding a target turf.
+	set waitfor = FALSE
+
+	while(pulses_remaining)
+		sleep(pulse_delay)
```

Comment:
Is this inside of a spawn, or will it hang everything?

Reply:
waitfor = FALSE sets it so that it's functionally in a spawn